src/solver/solve.jl

```diff
@@ -196,3 +196,69 @@ function precompile_callbacks(integrator)
     @assert B
     return nothing
 end
+
+
+check_conservation(simulation::AtmosSimulation) =
+    check_conservation(simulation.integrator.sol)
+check_conservation(integrator::CTS.DistributedODEIntegrator) =
+    check_conservation(integrator.sol)
+check_conservation(atmos_sol::AtmosSolveResults) =
+    check_conservation(atmos_sol.sol)
+
+"""
+    check_conservation(sol)
+    check_conservation(simulation)
+    check_conservation(integrator)
+
+Return:
+- `energy_conservation = energy_net / energy_total`
+- `mass_conservation = (mass(t_end) - mass(t_0)) / mass(t_0)`
+- `water_conservation = (water_atmos + water_surface) / water_total`
+
+"""
+function check_conservation(sol)
+    # energy
+    energy_total = sum(sol.u[end].c.ρe_tot)
+    energy_atmos_change = sum(sol.u[end].c.ρe_tot) - sum(sol.u[1].c.ρe_tot)
+    p = sol.prob.p
+    sfc = p.atmos.surface_model
+    if sfc isa PrognosticSurfaceTemperature
+        sfc_cρh = sfc.ρ_ocean * sfc.cp_ocean * sfc.depth_ocean
+        energy_total +=
+            horizontal_integral_at_boundary(sol.u[end].sfc.T .* sfc_cρh)
+        energy_surface_change =
+            horizontal_integral_at_boundary(
+                sol.u[end].sfc.T .- sol.u[1].sfc.T,
+            ) * sfc_cρh
+    else
+        energy_surface_change = -p.net_energy_flux_sfc[][]
+    end
+    energy_radiation_input = -p.net_energy_flux_toa[][]
+
+    energy_net =
+        abs(
+            energy_atmos_change + energy_surface_change -
+            energy_radiation_input,
+        ) / energy_total
```

Comment:
```suggestion
    energy_conservation =
        abs(
            energy_atmos_change + energy_surface_change -
            energy_radiation_input,
        ) / energy_total
```
(I think there is something wrong with the conservation test before? We are dividing it by energy_total here, but then divide it by energy_total again for energy_conservation.)

Reply:
Yes, I think you are right